### PR TITLE
Licence relation report optimization

### PR DESCRIPTION
**Pull request by szok**

### CODE REVIEW

File: src/ralph/reports/views.py [hunk 1]
```diff
@@ -360,38 +359,54 @@ class LicenceRelationsReport(BaseRelationsReport):
...
+    ]
 
     def prepare(self, model, *args, **kwargs):
         queryset = Licence.objects.all()
         if model._meta.object_name == 'BackOfficeAsset':
             queryset = queryset.filter(
-                base_objects__content_type=ContentType.objects.get_for_model(
-                    BackOfficeAsset
-                )
+                software__asset_type=ObjectModelType.back_office
```
Comment (mkurek) [inline src/ralph/reports/views.py]: `software__asset_type__in=(ObjectModelType.back_office, ObjectModelType.all)`

File: src/ralph/reports/tests.py [hunk 2]
```diff
@@ -177,6 +178,13 @@ def test_asset_relation(self):
         ]
         self.assertEqual(report_result, result)
 
+    def test_num_queries(self):
```
Comment (mkurek) [inline src/ralph/reports/tests.py]: could you make the same test for BackOffice?

Comment (mkurek) [inline src/ralph/reports/tests.py]: btw in this kind of test it'd be cool to provide sufficient amount of test data (ex. 1000 licences)

**Comment by mkurek** [comment]
:ok:

File: src/ralph/reports/tests.py [hunk 3]
```diff
@@ -155,11 +159,35 @@ def setUp(self):
...
+                niw='N/A' + str(i),
+                software__name=str(i)
+            )
+            BaseObjectLicence.objects.create(
+                licence=licence, base_object=bo.baseobject_ptr
+            )
+            LicenceUser.objects.create(
+                licence=licence,
+                user=user
+            )
```
Comment (ar4s) [inline src/ralph/reports/tests.py]: try `bulk_create`
https://docs.djangoproject.com/en/1.8/ref/models/querysets/#bulk-create

Comment (mkurek) [inline src/ralph/reports/tests.py]: :+1:

**Comment by ar4s** [comment]
:ok:

**Pull request merged**